daemons/gptp/common/ptp_message.cpp

```diff
@@ -959,9 +962,8 @@ void PTPMessageFollowUp::processMessage(IEEE1588Port * port)
 	master_local_freq_offset += 1.0;
```

Comment:
Exactly.  There is much subtlety required to do all these calculations correctly with minimal loss of precision.  We're not quite there yet, apparently, but this patch should definitely help.